56 contour instances

ronsilverentand opened the pull request:
Added tests for all Contour classes and modified classes to pass tests

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **Refactor**
  - Improved dynamic subscription handling and refined data conversion behavior for lists, objects, and booleans with enhanced error feedback.

- **New Features**
  - Introduced additional methods for managing object instances.
  - Outlined plans for future numeric precision and integer handling improvements.

- **Tests**
  - Expanded the test suite to robustly cover variable instances, key manipulation, type coercion, and subscription mechanisms.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ronsilverentand committed:
Testing added

Added tests and made object updatable with brackets syntax

lucasilverentand committed:
added todos for contour

lucasilverentand commented:
@coderabbit review

lucasilverentand commented on packages/contour/lib/src/types/object.dart:
```diff
@@ -183,22 +188,29 @@ class ObjectVariableInstance extends VariableInstance<Map<String, dynamic>> {
-
-//   User(this.name, this.age);
-// }
+  /// Modified operator [] to return the instance for compound types.
+  dynamic operator [](String key) {
+    final instance = _instances[key];
+    if (instance == null) return null;
+    if (instance is ObjectVariableInstance ||
+        instance is ListVariableInstance) {
+      return instance;
+    }
+    return instance.value;
+  }
```
This will make the API more confusing

ronsilverentand committed:
All issues fixed plus additional tests

tests added for functional validation of equals, optional, required, fallback, OneOf

lucasilverentand commented on packages/contour/test/src/instance_test.dart:
```diff
@@ -0,0 +1,38 @@
+      expect(notified, isTrue);
+    });
+
+    test('should update errors and validity when value is invalid', () {
+      final instance = SingleVariableInstance(
+        ContourString().equals('expected'),
+        'test',
+      );
+      instance.value = 'unexpected';
+      expect(instance.value, isNull);
+      expect(instance.errors, isNotEmpty);
+      expect(instance.valid, isFalse);
+    });
```
This should be a separate issue

lucasilverentand committed:
Refactor key shifting logic and improve coercion handling in Contour types; enhance tests for segment handling and error reporting

lucasilverentand reviewed (approved):
All good
